src/tdastro/obstable/roman_obstable.py

```diff
@@ -0,0 +1,249 @@
+        self.ma_table = _get_ma_table(ma_table_path)
+        roman_char = _get_roman_char()
+        self.zp_table = roman_char["zp_table"]
+        self.psf_table = roman_char["psf_table"]
+        self.thermal_table = roman_char["thermal_table"]
+        self.zodiacal_table = roman_char["zodiacal_min_table"]
+        self._append_apt_table()
+
+        super().__init__(self.apt_table, colmap=colmap, **kwargs)
+
+    def _append_apt_table(self):
+        self.apt_table["zp_abmag"] = 0.0
+        self.apt_table["N_Eff_Pix"] = 0.0
+        self.apt_table["zodi_countrate_min"] = 0.0
+        self.apt_table["thermal_countrate"] = 0.0
+        self.apt_table["time"] = 0.0
+        self.apt_table["exptime"] = 0.0
+
+        for f in np.unique(self.apt_table.BANDPASS):
+            zp_abmag = self.zp_table.loc[self.zp_table.element == f, "ABMag"].values[0]
+            self.apt_table.loc[f == self.apt_table.BANDPASS, "zp_abmag"] = zp_abmag
+            n_eff_pix = self.psf_table.loc[self.psf_table["Filter"] == f, "N_Eff_Pix"].values[0]
+            self.apt_table.loc[f == self.apt_table.BANDPASS, "N_Eff_Pix"] = n_eff_pix
+            sigma_zodi_min = self.zodiacal_table.loc[self.zodiacal_table["filter"] == f, "rate"].values[0]
+            self.apt_table.loc[f == self.apt_table.BANDPASS, "zodi_countrate_min"] = sigma_zodi_min
+            sigma_thermal = self.thermal_table.loc[self.thermal_table["filter"] == f, "rate"].values[0]
+            self.apt_table.loc[f == self.apt_table.BANDPASS, "thermal_countrate"] = sigma_thermal
+
+        for mat_number in np.unique(self.apt_table.MA_TABLE_NUMBER):
+            exptime = self.ma_table.loc[self.ma_table["MATableNumber"] == mat_number, "Exptime"].values[0]
+            self.apt_table.loc[mat_number == self.apt_table.MA_TABLE_NUMBER, "exptime"] = exptime
+
+        npass = len(np.unique(self.apt_table.PASS))
+        self.start_time = 61304.0
+        self.cadence = 5.0
+        times = self.start_time + self.cadence * np.arange(npass)
+        for i, pass_num in enumerate(np.unique(self.apt_table.PASS)):
+            self.apt_table.loc[pass_num == self.apt_table.PASS, "time"] = times[i]
+
+        self.apt_table["zp_nJy"] = mag2flux(self.apt_table["zp_abmag"])
+
+    def readnoise_func(self, exptime):
+        """
+        Readout noise function for Roman.
+        Reference: Eq. 9 of Rose et al 2025.
+
+        Parameters
+        ----------
+        exptime: float or npt.ArrayLike
+            Exposure time in seconds.
+
+        Returns
+        -------
+        sigma_read: float or npt.ArrayLike
+            Readout noise.
+        """
+        exptime = np.asarray(exptime)
+        sigma_floor_sq = 25.0
+        sigma_sq = 3072.0
+        n = exptime / 3.04
+        nfactor = (n - 1) / n / (n + 1)
+        sigma_read = np.sqrt(sigma_floor_sq + sigma_sq * nfactor)
+
+        return sigma_read
+
+    def calculate_skynoise(self, exptime, zodi_scale, zodi_countrate_min, thermal_countrate):
```

Comment:
If you wanted to, you could at an `@cite_function` decorator to pick up the reference in the docstring automatically. (You'd also need `from citation_compass import cite_function` at the top).